### DISCUSSION TITLE
Using in Drupal 8/9

### DISCUSSION THREAD

**Discussion by ericpugh** [spatie/drupal-ray#3] [General]
Hello,

If helpful to anyone, I just created a [drupal module](https://github.com/Smithsonian/Drupal-Ray-Debugger) to integrate Ray in their Drupal projects. The module doesn't really _do_ anything, of course, just adds dependencies for the Ray "framework agnostic project", javascript, etc and a simple Twig extension in a way that Drupal expects. 

I would be interested if any other Drupal folks were using Ray or if Spatie had any plans for _this_ package. For example, let's say I wanted to be able to use `showQueries()` the way the Laravel package does, could maybe get something working in my module as a [macro](https://spatie.be/docs/ray/v1/advanced-usage/adding-your-own-ray-functions-in-php) but maybe it would be better to work on a specific package for Drupal, and leave some of this "integration"/setup in a separate module or up to the individual project.

Thoughts?

**Comment by freekmurze** [comment]
Hi Eric, this looks great! I unfortunately don't have any Drupal experience, so I can't help you technically with this.

I could however give you admin access to this repo https://github.com/spatie/drupal-ray in which you could copy and develop your module. Currently that repo holds just some scaffolded files that can be replaced.

I feel like a first party package might be more thrusthworthy for others to use (although it living on the Smithsonian repo is pretty cool 🙂) We could also add a link to the Drupal module in the docs when it's in a more finished state.

Let me know if you want any of this to happen 🙂 In any case, thanks for your work!

**Reply by ericpugh** [reply]
Hi, I actually decided to move it to a [drupal.org project](https://www.drupal.org/project/ray_debugger/). This is just the logical place that other Drupal devs will expect to find it, report issues, etc. (Note: the scary looking "security" banner is a temporary thing because it's a brand new project). Feel free to point users to it. It would be interesting to hear from some other Drupal devs because Ray really _is_ a useful tool... even in the strange world of Drupal. :smile: 

thanks!

**Reply by freekmurze** [reply]
I've added a link to the model in our docs: https://spatie.be/docs/ray/v1/installation-in-your-project/drupal

If there are any Drupal specific methods, use the edit button on this page to add them to our docs: https://spatie.be/docs/ray/v1/usage/drupal
